worker/pruner/worker.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package pruner
+
+import (
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/clock"
+
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/watcher"
+	"github.com/juju/juju/worker/catacomb"
+)
+
+var logger = loggo.GetLogger("juju.worker.pruner")
+
+// Facade represents an API that implements status history pruning.
+type Facade interface {
+	Prune(time.Duration, int) error
+	WatchForModelConfigChanges() (watcher.NotifyWatcher, error)
+	ModelConfig() (*config.Config, error)
+}
+
+//updates model configuration for any implementation of a pruner worker
+type ModelConfigUpdater interface {
+	MaxEntryAgeUpdate() time.Duration
```

Comment:
I don't understand why these methods end with Update. The implementations of them are just getters.